### ISSUE TITLE
document_package option for building R packages

### ISSUE THREAD

**Issue by markvanderloo** [travis-ci/travis-ci#4081]
Dear  @craigcitro, @eddelbuettel and @hadley,

I try to avoid adding generated files like ```Rd``` and ```NAMESPACE``` to my github repos, since these are generated by ```roxygen2```. If I understand correctly, using the ```language r``` directive in ```.travis.yml``` it is not possible to include a step that generates documentation. 

I tried ```before_install``` and ```after_install```, with no joy.

I succeeded in building documentation on the travis service after several commits (see [here](https://github.com/markvanderloo/stringdist/blob/master/.travis.yml)) by using the more basic commands and ```travis-tool.sh```. 

My point is: it may be nice to have something like a ```before_build``` option that allows users to insert documentation procedures before building the package, but after installing all dependencies. For example:

```
before_build:
 - roxygenize('.')
```
But please correct me if this is already possible somehow: I'm not that savvy on ```travis-ci```

Thanks,
Mark

**Comment by eddelbuettel** [comment]
No idea.  FWIW I do install `.Rd` from `roxygen2` in my packages,

**Comment by krlmlr** [comment]
Have you tried:

```
before_script: Rscript -e "roxygen2::roxygenize()"
```

This seems to work for me ([build](https://travis-ci.org/krlmlr/autoroxy/builds/105409787), [code](https://github.com/krlmlr/autoroxy/blob/a9d7e626d8bbbf1a0c9939d072ba16694381f963/.travis.yml#L7-L8)). The autoroxy package offers a way to turn on and off the automatic generation of documentation, too.

**Comment by eddelbuettel** [comment]
I tend to have [this helper script](https://github.com/eddelbuettel/littler/blob/master/inst/examples/roxy.r) in my `$PATH`.

We need to get R Core to give us a 'hook' to run this, `Rcpp::compileAttributes()` and other generators.  So far I can only think of abusing `cleanup` or `configure`.

**Comment by krlmlr** [comment]
I remember that configure requires Rtools on the OS which you don't usually seem to care about.

FWIW, hooking into `R CMD INSTALL` does the job for documentation (but not for creating NAMESPACE); I wonder if this could work for compileAttributes(), too.

**Comment by eddelbuettel** [comment]
I don't think it does. `configure` can anything with mode 0755, similarly for `configure.win`. Show me a line in the manuals that states this.  I *think* I have counterexamples eg in package Rblpapi.

[ That said, Windows systems are close to brain-dead for development, and Windows systems without Rtools are useful mostly for playing Solitaire so personally I don't give a rat's arse about them.  But what I said passes muster with `R CMD check` which is really all we care about. ]

**Comment by krlmlr** [comment]
The [code](https://github.com/wch/r-source/blob/9ab83101d3cc2d64f88caa4f65b3337ab0006e74/src/library/tools/R/install.R#L624-L634) shows that, on Windows, if `configure.win` exists, it is executed using `sh`. This is for `R CMD INSTALL`; I got a warning trying to run it on a fresh clone of your Rblpapi package.

Which R core member would support such a hook for generators?

**Comment by eddelbuettel** [comment]
Ack.  So configure.win would require Rtools.  

I do not know who we could work with on this, and I very doubt it would remove the dependency you appear to dread here.
